### PR TITLE
Infrastructure

### PR DESCRIPTION
**Pull request by Shaktam**
except patent_watch.tf

#16 #17 #18

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+infrastructure/patent_watch.tf
```
Comment (fabianschmauder) [inline .gitignore]: empty line :)

Comment (fabianschmauder) [inline .gitignore]: Why you need to ignore this file?

File: infrastructure/.gitignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+.terraform
+.terraform.lock.hcl
+terraform.tfstate
+terraform.tfstate.backup
+infrastructure/patent_watch.tf
```
Comment (fabianschmauder) [inline infrastructure/.gitignore]: you ignore the file twice

Comment (fabianschmauder) [inline infrastructure/.gitignore]: empty line

File: infrastructure/autoscaling.tf [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  
+  instance_type = "t3.micro"
+  
+  key_name = "vockey"
+
+
+  vpc_security_group_ids = [aws_security_group.security_group.id]
+
+
+ # user_data = filebase64("script/userdata.sh")
```
Comment (fabianschmauder) [inline infrastructure/autoscaling.tf:19]: if you do not need the line i would remove it

File: infrastructure/dynamodb.tf [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  stream_enabled   = true
+  stream_view_type = "NEW_AND_OLD_IMAGES"
+  read_capacity    = 25
+  write_capacity   = 25
+
+  attribute {
+    name = "name"
+    type = "S"
+  }
+}
```
Comment (fabianschmauder) [inline infrastructure/dynamodb.tf]: empty line

File: infrastructure/lambda.tf [hunk 5]
```diff
@@ -0,0 +1,31 @@
+resource "aws_lambda_function" "patent_lambda" {
+  filename      = "patent_informer.zip"
+  function_name = "patent-notifier-dynamodb"
+  role          = "arn:aws:iam::467749311079:role/LabRole"
```
Comment (fabianschmauder) [inline infrastructure/lambda.tf:4]: it is better to use the account id from the data 
[documentation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/caller_identity)

than you can join it with "arn:aws:iam::" and ":role/LabRole"

File: infrastructure/lambda.tf [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  handler       = "patent-dynamodb.handler"
+  timeout       = 300
+  runtime       = "python3.9"
+  source_code_hash = filebase64sha256("x.zip")
```
Comment (fabianschmauder) [inline infrastructure/lambda.tf:8]: where is the code?

File: infrastructure/lambda.tf [hunk 7]
```diff
@@ -0,0 +1,31 @@
+resource "aws_lambda_function" "patent_lambda" {
+  filename      = "patent_informer.zip"
```
Comment (fabianschmauder) [inline infrastructure/lambda.tf]: filename is different to the source_code_hash?

File: infrastructure/lambda.tf [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+resource "aws_sns_topic" "patent_email" {
+  name = "Patent-sns-Notification"
+}
+
+resource "aws_sns_topic_subscription" "patent_update_notification" {
+  topic_arn = aws_sns_topic.patent_email.arn
+  protocol  = "email"
+  endpoint  = "cuteshakti1493@gmail.com"
+}
```
Comment (fabianschmauder) [inline infrastructure/lambda.tf]: The file is called lambda but you also have sns in your file

File: infrastructure/patent_watch.tf [hunk 9]
```diff
@@ -0,0 +1,45 @@
+resource "aws_lambda_function" "patent_notifier" {
```
Comment (fabianschmauder) [inline infrastructure/patent_watch.tf:1]: file in git ignore but in your pr?

File: infrastructure/patent_watch.tf [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  function_name = "patent_notifier_s3"
+  filename       = "build/job_crawler.zip"
+  role          = "arn:aws:iam::467749311079:role/LabRole"
```
Comment (fabianschmauder) [inline infrastructure/patent_watch.tf:4]: you can move the role to a variable. Than it is easier to change it later

File: infrastructure/patent_watch.tf [hunk 11]
```diff
@@ -0,0 +1,45 @@
+resource "aws_lambda_function" "patent_notifier" {
+  function_name = "patent_notifier_s3"
+  filename       = "build/job_crawler.zip"
```
Comment (fabianschmauder) [inline infrastructure/patent_watch.tf]: where is the code?

File: infrastructure/patent_watch.tf [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+  role          = "arn:aws:iam::467749311079:role/LabRole"
+  handler       = "patent-s3.handler"
+  timeout       = 300
+  runtime       = "python3.9"
+  layers        = [aws_lambda_layer_version.requests_layer.arn]
+  source_code_hash = filebase64sha256("patent_notifier.zip")
```
Comment (fabianschmauder) [inline infrastructure/patent_watch.tf:9]: different to "build/job_crawler.zip"

**Review by fabianschmauder — COMMENTED** [review]
This Pr is not really a feature.

It is quite huge and has no code in it.

The rest looks fine

File: .gitignore (hunk 1 above)
Comment (Shaktam) [reply .gitignore]: I just created this file to check my code using s3 bucket before using in dynamodb after I get my api

**Pull request merged**